src/amo/components/AddonDescription.js

```diff
@@ -0,0 +1,86 @@
+import classNames from 'classnames';
+import React, { PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+import { compose } from 'redux';
+
+import { nl2br, sanitizeHTML } from 'core/utils';
+import translate from 'core/i18n/translate';
+import Card from 'ui/components/Card';
+
+import './AddonDescription.scss';
+
+
+export const allowedDescriptionTags = [
+  'a',
+  'abbr',
+  'acronym',
+  'b',
+  'blockquote',
+  'br',
+  'code',
+  'em',
+  'i',
+  'li',
+  'ol',
+  'strong',
+  'ul',
+];
+
+export class AddonDescriptionBase extends React.Component {
+  static propTypes = {
+    addon: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { expanded: false };
+  }
+
+  componentDidMount() {
+    this.expandIfDescriptionIsShortEnough();
+  }
+
+  expandIfDescriptionIsShortEnough = () => {
+    // If the add-on description is short enough it doesn't need a "show more"
+    // link, we'll expand the description by default.
+    if (ReactDOM.findDOMNode(this.contents).clientHeight < 100) {
+      this.setState({ expanded: true });
+    }
+  }
+
+  expandText = (event) => {
+    event.preventDefault();
+    this.setState({ expanded: true });
+  }
+
+  render() {
+    const { addon, i18n } = this.props;
+    const { expanded } = this.state;
+
+    const header = i18n.sprintf(
+      i18n.gettext('About this %(addonType)s'), { addonType: addon.type });
+
+    /* eslint-disable react/no-danger */
+    return (
+      <Card className={classNames('AddonDescription', {
+        'AddonDescription--expanded': expanded,
+      })} header={header}>
+        <div className="AddonDescription-contents"
+          ref={(ref) => { this.contents = ref; }}
+          dangerouslySetInnerHTML={
+            sanitizeHTML(nl2br(addon.description), allowedDescriptionTags)
+          } />
+        <a className="AddonDescription-revealMoreLink" href="#more"
+          onClick={this.expandText}>
+          {i18n.gettext('Show more info…')}
```

Comment:
I feel like "Show more" might be better text since there's a section called "More information".